src/tests/Microsoft.PowerFx.Interpreter.Tests/DatabaseTests.cs

```diff
@@ -0,0 +1,212 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Interpreter.Tests
+{
+    public class DatabaseTests
+    {
+        [Fact]
+        public void PatchFirst()
```

Comment:
>PatchFirst [](http://example.com/codeflow?start=20&length=10)

test matrix: 2*2 = 4 cases:

- arg0 Table - "dataverse table" vs. in-memory table 
dv table means TableValue overrides key virtuals 

- arg1 baseRecord - dataverse record vs. in-memory record.
dv record means override TryGetPrimaryKey

Reply:
Also:

// We may need clarification on execptation here, when baseRecord includes another field that doesn't match
Patch( x, { contactid: x.contactid, address1_city: "othercity"} , {address1_city: "Chicago"} )

Reply:
Test matrix -> Patch.txt is covering most combinations
Asked Greg about the clarification

Reply:
But Patch.txt is not covering them with this combination of overloads (TestDatabaseRecordValue)